KachnaOnline.App/Controllers/EventsController.cs

```diff
@@ -0,0 +1,237 @@
+    public class EventsController : ControllerBase
+    {
+        private readonly EventsFacade _eventsFacade;
+
+        public EventsController(EventsFacade eventsFacade)
+        {
+            _eventsFacade = eventsFacade;
+        }
+
+        /// <summary>
+        /// Returns the list of events being held at the specified time.
+        /// </summary>
+        /// <param name="at">Specifies the date and time when the events have to be held.</param>
+        /// <returns>A List of <see cref="KachnaOnline.Dto.Events.EventDto"/> corresponding to the <paramref name="at"/> date and time.</returns>
+        /// <response code="200">The list of events.</response>
+        [AllowAnonymous]
+        [HttpGet("at/{at}")]
+        [ProducesResponseType(typeof(IEnumerable<EventDto>), 200)]
+        [ProducesResponseType(404)]
+        public async Task<ActionResult<IEnumerable<EventDto>>> GetEvents(DateTime at)
+        {
+            var events = await _eventsFacade.GetEvents(at);
+            return new ActionResult<IEnumerable<EventDto>>(events);
+        }
+
+        /// <summary>
+        /// Returns an event with the given ID.
+        /// </summary>
+        /// <param name="eventId">ID of the event to return.</param>
+        /// <returns>An <see cref="KachnaOnline.Dto.Events.EventDto"/> of a game corresponding to ID <paramref name="eventId"/>.</returns>
+        /// <response code="200">The event.</response>
+        /// <response code="404">No such event exists.</response>
+        [AllowAnonymous]
+        [HttpGet("{eventId}")]
+        [ProducesResponseType(typeof(EventDto), 200)]
+        [ProducesResponseType(404)]
+        public async Task<ActionResult<EventDto>> GetEvent(int eventId)
+        {
+            try
+            {
+                return await _eventsFacade.GetEvent(eventId);
+            }
+            catch (EventNotFoundException)
+            {
+                return this.NotFound();
+            }
+        }
+
+        /// <summary>
+        /// Returns the list of current events.
+        /// </summary>
+        /// <returns>A List of <see cref="KachnaOnline.Dto.Events.EventDto"/> of the current events.</returns>
+        /// <response code="200">The list of current events.</response>
+        [AllowAnonymous]
+        [HttpGet("current")]
+        [ProducesResponseType(typeof(IEnumerable<EventDto>), 200)]
+        public async Task<ActionResult<IEnumerable<EventDto>>> GetCurrentEvents()
+        {
+            var events = await _eventsFacade.GetCurrentEvents();
+            return new ActionResult<IEnumerable<EventDto>>(events);
+        }
+
+        /// <summary>
+        /// Returns the list of next planned events.
+        /// </summary>
+        /// <returns>A List of <see cref="KachnaOnline.Dto.Events.EventDto"/> of the next planned events.</returns>
+        /// <response code="200">The list of the next planned events.</response>
+        [AllowAnonymous]
+        [HttpGet("next")]
```

Comment:
No, it is truly returning only the next event(s) – the first event(s) to come – but because events can overlap, it is possible two events start at the same time and therefore should both be returned. It might never happen, but for the matter of precision, we should remember the system allows this.